### PR TITLE
Dev tools and Dapp example tests Fix

### LINKED ISSUE #619: Bump local node dapp and tool test images for 0.31.A
### Problem

Currently the dapp and tool tests are using images older than 0.31.A

### Solution

Bump local node dapp and tool test images for 0.31.A

### Alternatives

_No response_

### LINKED ISSUE #638: Truffle dev tool broken on PollingBlockTracker
### Description

Recently the truffle dev tools stopped working and started to fail with 

```
/home/runner/work/hedera-json-rpc-relay/hedera-json-rpc-relay/tools/truffle-example/node_modules/eth-block-tracker/src/polling.js:51
        const newErr = new Error(`PollingBlockTracker - encountered an error while attempting to update latest block:\n${err.stack}`)
                       ^
Error: PollingBlockTracker - encountered an error while attempting to update latest block:
undefined
    at PollingBlockTracker._performSync (/home/runner/work/hedera-json-rpc-relay/hedera-json-rpc-relay/tools/truffle-example/node_modules/eth-block-tracker/src/polling.js:51:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
UnhandledRejections detected
Promise {
  <rejected> {
    code: -32603,
    message: 'connect ECONNREFUSED [12](https://github.com/hashgraph/hedera-json-rpc-relay/actions/runs/3278788343/jobs/5397619132#step:11:13)7.0.0.1:7546',
    data: { originalError: [Object] }
  }
}
```

See https://github.com/hashgraph/hedera-json-rpc-relay/actions/workflows/dev-tool-test.yml?query=is%3Afailure for recent failures

### Steps to reproduce

Follow [Truffle example steps](https://github.com/hashgraph/hedera-json-rpc-relay/tree/main/tools/truffle-example)

### Additional context

_No response_

### Hedera network

other

### Version

v0.10.0

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
Signed-off-by: georgi-l95 <glazarov95@gmail.com>

**Description**:

- Bump hardhat version in dev tools
- Bump truffle version in dev tools
- Bump network and mirror node image in dapp/dev tool tests and acceptance tests
- Add check if local node is started, before continuing with the workflow
- Dapp workflow change, check if local node is started before bootstrap script
- Bump @hashgraph/hedera-local version everywhere

**Related issue(s)**:

Fixes #638 
Fixes #619 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by lukelee-sl — APPROVED** [review]

File: packages/server/tests/helpers/nodeCheck.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                console.log(
+                `Waiting for local node, retrying in 15 seconds...`
+                );
+            });
+
+            await new Promise(r => setTimeout(r, 15000));
+        }
+    }
+    process.exit(0);
+})();
```
Comment (Nana-EC) [inline packages/server/tests/helpers/nodeCheck.ts:28]: nit: new line for easier git diff

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -503,6 +503,13 @@ export class EthImpl implements Eth {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     this.logger.trace(`${requestIdPrefix} getBalance(account=${account}, blockNumberOrTag=${blockNumberOrTag})`);
 
+    if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:506]: Q: What's the need for this?
Add a comment.

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -503,6 +503,13 @@ export class EthImpl implements Eth {
...
+      const latestBlock = await this.blockNumber(requestId);
+      if (latestBlock === blockNumberOrTag) {
+        blockNumberOrTag = latestBlock
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:509]: ALso if not already covered please add a test case in `eth.spec.ts` to make sure the call made to the mirror node is as expected.

**Review by Nana-EC — DISMISSED** [review]
LG, some nits

File: packages/relay/src/lib/eth.ts (hunk 2 above)
Comment (Nana-EC) [reply packages/relay/src/lib/eth.ts:506]: Nice, good find.

Comment (Nana-EC) [reply packages/relay/src/lib/eth.ts:509]: Also created this ticket which will address this issue https://github.com/hashgraph/hedera-mirror-node/issues/4762
Hopefully we can focus on this in the next sprint

**Review by Nana-EC — APPROVED** [review]
LG

File: packages/relay/src/lib/eth.ts (hunk 2 above)
Comment (georgi-l95) [reply packages/relay/src/lib/eth.ts:509]: Nice

**Pull request merged**